editor/src/clj/editor/camera_editor.clj

```diff
@@ -205,115 +213,46 @@
-    (set! (. m m11) (/ 2.0 (- top bottom)))
-    (set! (. m m12) 0.0)
-    (set! (. m m13) (/ (- (+ top bottom)) (- top bottom)))
-
-    (set! (. m m20) 0.0)
-    (set! (. m m21) 0.0)
-    (set! (. m m22) (/ 2.0 (- near far)))
-    (set! (. m m23) (/ (+ near far) (- near far)))
-
-    (set! (. m m30) 0.0)
-    (set! (. m m31) 0.0)
-    (set! (. m m32) 0.0)
-    (set! (. m m33) 1.0)
-    m))
-
-(defn- camera-projection-matrix [is-orthographic near far fov]
-  (if (true? is-orthographic)
-    (camera-orthographic-projection-matrix near far fov)
-    (camera-perspective-projection-matrix near far fov)))
+(defn- camera-projection-matrix [is-orthographic near far fov-deg aspect-ratio]
+  ;; TODO: Derive aspect-ratio from display setting in game.project when auto-aspect-ratio is enabled.
+  (let [fov-y (double fov-deg)
+        fov-x (* fov-y (double aspect-ratio))]
+    (if (true? is-orthographic)
+      (camera/simple-orthographic-projection-matrix near far fov-x fov-y)
+      (camera/simple-perspective-projection-matrix near far fov-x fov-y))))
 
 (defn- gen-outline-vertex-buffer
-  [renderables count]
-  (let [camera-mesh-vertices-count 212 ;(count camera-mesh-lines) ; why doesn't this work?
-        color-vertices-count (+ camera-mesh-vertices-count 32)]
-    (loop [renderables renderables
-           vbuf (->color-vtx (* count color-vertices-count))]
-      (if-let [renderable (first renderables)]
-        (let [color (if (:selected renderable) colors/selected-outline-color colors/outline-color)
-              cr (get color 0)
-              cg (get color 1)
-              cb (get color 2)
-              user-data (:user-data renderable)
-              world-translation (:world-translation renderable)
-              world-rotation (:world-rotation renderable)
-              ^Matrix4d proj-matrix (camera-projection-matrix (:is-orthographic user-data) (:near-z user-data) (:far-z user-data) (:fov user-data))
-              ^Matrix4d view-matrix (camera-view-matrix world-translation world-rotation)
-              ^Matrix4d inv-view-matrix (math/inverse view-matrix)
-              ^Matrix4d proj-view-matrix (doto (Matrix4d. proj-matrix) (.mul view-matrix))
-              ^Matrix4d inv-view-proj-matrix (math/inverse proj-view-matrix)]
-          (recur (rest renderables) (conj-camera-outline! vbuf world-translation inv-view-matrix inv-view-proj-matrix cr cg cb)))
-        (persistent! vbuf)))))
-
-(defn- render-frustum-outlines [^GL2 gl render-args renderables count]
+  [renderables ^long renderable-count]
+  (loop [renderables renderables
+         vbuf (->color-vtx (* renderable-count camera-preview-mesh-vertices-count))]
+    (if-let [renderable (first renderables)]
+      (let [color (if (:selected renderable) colors/selected-outline-color colors/outline-color)
+            cr (get color 0)
+            cg (get color 1)
+            cb (get color 2)
+            {:keys [is-orthographic near-z far-z fov aspect-ratio]} (:user-data renderable)
```

Comment:
I use [destructuring](https://clojure.org/guides/destructuring) here to unpack the `user-data`.